src/bpmncwpverify/bpmn.py

```diff
@@ -223,18 +246,33 @@ def accept(self, visitor: "BpmnVisitor") -> None:
 # Bpmn class (building graph from xml happens here)
 ###################
 class Bpmn:
-    _TAG_CLASS_MAPPING = {
+    _TAG_CLASS_MAPPING: Dict[str, Type[BpmnElement]] = {
         "task": Task,
         "startEvent": StartEvent,
         "endEvent": EndEvent,
         "exclusiveGateway": ExclusiveGatewayNode,
         "parallelGateway": ParallelGatewayNode,
+        "sendTask": IntermediateEvent,
+        "intermediateCatchEvent": IntermediateEvent,
+        "intermediateThrowEvent": IntermediateEvent,
     }
 
     _FLOW_MAPPING = {"sequenceFlow": SequenceFlow}
+    _MSG_MAPPING = {"messageFlow": MessageFlow}
 
     def __init__(self) -> None:
-        self.processes: List[Process] = []
+        self.processes: Dict[str, Process] = {}
+        self.id_to_element: Dict[str, BpmnElement] = {}  # Maps ids to elements
```

Comment:
this is the map that holds the ids to the elements. This is how we can map from promela back to the bpmn. Promela willjust use element's ids (which are already compliant with promela).

Reply:
Perfect.